js/plugins/vertexai/src/gemini.ts

```diff
@@ -594,19 +673,42 @@ export function defineGeminiModel(
   const model: ModelReference<z.ZodTypeAny> = SUPPORTED_GEMINI_MODELS[name];
   if (!model) throw new Error(`Unsupported model: ${name}`);
 
+  return defineGeminiModel(
+    ai,
+    modelName,
+    name,
+    model?.info,
+    vertexClientFactory,
+    options
+  );
+}
+
+/**
+ * Define a Vertex AI Gemini model.
+ */
+export function defineGeminiModel(
+  ai: Genkit,
+  modelName: string,
+  version: string,
+  modelInfo: ModelInfo | undefined,
+  vertexClientFactory: (
+    request: GenerateRequest<typeof GeminiConfigSchema>
+  ) => VertexAI,
+  options: PluginOptions
+): ModelAction {
   const middlewares: ModelMiddleware[] = [];
-  if (SUPPORTED_V1_MODELS[name]) {
+  if (!modelInfo?.supports?.systemRole) {
```

Comment:
I seem to recall we need to simulate system prompts for 1.0 era models, but we still want to say "systemRole=true" so that the dev ui will render the system prompt input, etc.

IOW, systemRole=true applies even to cases where its simulated.

Reply:
Separately, if we don't support 1.0 anymore, we can probably remove this?

Reply:
Good catch, we still need to support 1.0. Undone.